source/server/admin/prometheus_stats.cc

```diff
@@ -284,8 +316,16 @@ uint64_t PrometheusStatsFormatter::statsAsPrometheus(
   metric_name_count += outputStatType<Stats::TextReadout>(
       response, params, text_readouts, generateTextReadoutOutput, "gauge", custom_namespaces);
 
-  metric_name_count += outputStatType<Stats::ParentHistogram>(
-      response, params, histograms, generateHistogramOutput, "histogram", custom_namespaces);
+  // Due to current behavior around the default value of histogram_buckets, we must interpret the default "NoBuckets"
+  // differently from an explicit query parameter setting "NoBuckets".
+  if (Utility::queryParam(params.query_, "histogram_buckets").has_value() &&
```

Comment:
I think it'd be nicer to put all the semantics of parsing the query and using a different default bucket-mode into stats_params.cc rather than re-parsing it here.